src/AlternativesPerformance.h

```diff
@@ -0,0 +1,102 @@
+#ifndef AlternativesPerformance_H
+#define AlternativesPerformance_H
+
+#include "Criteria.h"
+#include "Performance.h"
+#include "PerformanceTable.h"
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <ostream>
+#include <string.h>
+#include <vector>
+
+class AlternativesPerformance : public PerformanceTable {
+public:
+  /**
+   * AlternativesPerformance standard constructor (PerformanceTable surcharged)
+   *
+   * @param perf_vect Vector of performance
+   * @param alt_assignment Map of alternative to the assigned category. Default
+   * alternatives assigned to empty categories
+   */
+  AlternativesPerformance(std::vector<Performance> &perf_vect,
+                          std::map<std::string, std::string> alt_assignment =
+                              std::map<std::string, std::string>{{"", ""}});
+
+  /**
+   * AlternativesPerformance constructor without perf values but set of
+   * criteria to evaluate performance over (PerformanceTable surcharged)
+   *
+   * @param nb_of_perfs Number of performance
+   * @param crits Criteria to evaluate performance over
+   * @param prefix Prefix to use for the name of each Performance created.
+   * Default = "alt"
+   * @param alt_assignment Map of alternative to the assigned category. Default
+   * alternatives assigned to empty categories
+   * */
+  AlternativesPerformance(int nb_of_perfs, Criteria &crits,
+                          std::string prefix = "alt",
+                          std::map<std::string, std::string> alt_assignment =
+                              std::map<std::string, std::string>{{"", ""}});
+
+  /**
+   * AlternativesPerformance constructor using an existing performance table
+   *
+   * @param alt_assignment Map of alternative assignements to categories
+   * @param perf_table Performance table to copy
```

Comment:
maybe have a constructor that gives as input an vector of alternative ids could be useful !
AA = AlternativeAssignment
ie AA(std::vector<std::string> ['a1,'a2'], int nbCategories) or/and AA(std::vector<std::string> ['a1,'a2'], Categories cat)

Reply:
Why use a vector and not use directly the hashmap: AA(std::map<std::string, std::string>{{"a1", ""},{"a2", ""}}?
About the category/categories object we should definitely discuss that!